### ISSUE TITLE
Eliminate "blacklist/whitelist" terminology in SelfControl UI

### ISSUE THREAD

**Issue by jkmaxwell** [SelfControlApp/selfcontrol#624]
blacklist -> block
whitelist -> allow

Thanks for prioritizing this.

**Comment by cstigler** [comment]
replying here as well [as Twitter](https://twitter.com/charliestigler/status/1270453856875339776) to keep the thread more clear:

thanks Justin for the note! I moved to "blocklist" for this reason a few years ago in most of the UI, but it still uses blacklist/whitelist in the blocklist editor UI. That's on the list to change in the next release of SelfControl - timing TBD since releases are generally infrequent. But probably a month or a few.

Terminology suggested has included blocklist / denylist instead of blacklist, and allowlist / safelist instead of whitelist. Slightly complicated by the fact that the app currently uses "blocklist" to mean _any_ kind of list, whether blocking or allowing sites, and this may mean making a few more strings dynamic that are currently static. If anybody has any thoughtful opinions, feel free to opine here :)

**Comment by slambert** [comment]
I may be in a too goofy mode here but maybe this can give some ideas:

- avoid list / embrace list
- repel / attract
- get out / come in

I also wonder if any of the translations we've included might provide some inspiration?

But the terms block, deny, allow, and safe all clear enough in my mind. The differences are minor.

**Comment by slambert** [comment]
It looks like others have walked this path before us...

https://github.com/rails/rails/issues/33677

Allow/Deny is what they decided on. Works for me!

**Comment by geoffruth** [comment]
From my perspective, I don't care tremendously about what the replacement terms are (both block/allow and deny/allow seem fine). Deploying an update soon around this feels more important -- I'd advocate for an update soon, even if it's solely to address this language.

Thanks for engaging so quickly around this. I really appreciate this.

**Comment by cstigler** [comment]
Hi all, just a heads-up that this change has been merged into code in [this commit](https://github.com/SelfControlApp/selfcontrol/commit/b1264dd2b53805a99bc07e87085cba3ea82f0080). It will go out in the next SC release, although that may still not be for a little while (given my slower pace of work on SC, I try to batch more changes into one release for efficiency).

A couple notes on the changes:
- I went for "blocklist" / "allowlist" - I think "deny" is nice for more-technical projects like rails, but thought block worked better for SelfControl since it's often described as a website blocker.
- I only changed the English localization, since I didn't want to make up translations on my own. That said, I welcome PRs from translators who can find a good translation for "blocklist"/"allowlist" in their region!
- I changed almost all references in the code at the same time. However, the settings variables `HostBlacklist` and `BlockAsWhitelist` have been left unchanged for now. Changing those would require a good bit of thought/testing to preserve backwards-compatibility; I don't think they're a priority for now.

Closing this issue now. Thanks for the engagement all!
